pkg/cloudprovider/aws/amifamily/bootstrap/bottlerocketsettings.go

```diff
@@ -14,110 +14,87 @@ limitations under the License.
-	NTP               *ntp               `toml:"ntp,omitempty"`
-}
-
-// kubernetes specific configuration for bottlerocket api
-type kubernetes struct {
-	APIServer                 string               `toml:"api-server"`
-	ClusterCertificate        *string              `toml:"cluster-certificate"`
-	ClusterName               *string              `toml:"cluster-name"`
-	ClusterDNSIP              *string              `toml:"cluster-dns-ip,omitempty"`
-	NodeLabels                map[string]string    `toml:"node-labels,omitempty"`
-	NodeTaints                map[string][]string  `toml:"node-taints,omitempty"`
-	MaxPods                   *int                 `toml:"max-pods,omitempty"`
-	StaticPods                map[string]staticPod `toml:"static-pods,omitempty"`
-	EvictionHard              map[string]string    `toml:"eviction-hard,omitempty"`
-	KubeReserved              map[string]string    `toml:"kube-reserved,omitempty"`
-	SystemReserved            map[string]string    `toml:"system-reserved,omitempty"`
-	AllowedUnsafeSysctls      []string             `toml:"allowed-unsafe-sysctls,omitempty"`
-	ServerTLSBootstrap        *bool                `toml:"server-tls-bootstrap,omitempty"`
-	RegistryQPS               *int                 `toml:"registry-qps,omitempty"`
-	RegistryBurst             *int                 `toml:"registry-burst,omitempty"`
-	EventQPS                  *int                 `toml:"event-qps,omitempty"`
-	EventBurst                *int                 `toml:"event-burst,omitempty"`
-	KubeAPIQPS                *int                 `toml:"kube-api-qps,omitempty"`
-	KubeAPIBurst              *int                 `toml:"kube-api-burst,omitempty"`
-	ContainerLogMaxSize       *string              `toml:"container-log-max-size,omitempty"`
-	ContainerLogMaxFiles      *int                 `toml:"container-log-max-files,omitempty"`
-	CPUManagerPolicy          *string              `toml:"cpu-manager-policy,omitempty"`
-	CPUManagerReconcilePeriod *string              `toml:"cpu-manager-reconcile-period,omitempty"`
-	TopologyManagerScope      *string              `toml:"topology-manager-scope,omitempty"`
-	TopologyManagerPolicy     *string              `toml:"topology-manager-policy,omitempty"`
+// BottlerocketConfig is the root of the bottlerocket config, see more here https://github.com/bottlerocket-os/bottlerocket#using-user-data
+type BottlerocketConfig struct {
+	SettingsRaw map[string]interface{} `toml:"settings"`
+	Settings    BottlerocketSettings   `toml:"-"`
 }
 
-type containerRegistry struct {
-	Credentials []*credential `toml:"credentials,omitempty"`
+// BottlerocketSettings is a subset of all configuration in https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/models/src/aws-k8s-1.22/mod.rs
+// These settings apply across all K8s versions that karpenter supports.
+type BottlerocketSettings struct {
+	Kubernetes BottlerocketKubernetes `toml:"kubernetes"`
```

Comment:
I was curious why we've only modeled `settings.kubernetes` and we're letting the others flow through to Bottlerocket (which probably does its own validation on supported fields)?

Reply:
I only modeled settings.kubernetes because those are the only ones we use in the bootstrap script since we override the endpoint and what not.  I have an issue out to support partial parsing in the go-toml https://github.com/pelletier/go-toml/issues/796

Reply:
Oh right that makes sense! Thanks